src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -129,49 +140,61 @@
            :else [:div {:style {:textAlign "center"}}
                   [comps/Spinner {:text "Loading Method Configuration..."}]]))
    :component-did-mount
-   (fn [{:keys [props state this]}]
-     (this :-load-validated-method-config)
-     (endpoints/call-ajax-orch
-      {:endpoint endpoints/list-methods
-       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                  (if success?
-                    (swap! state assoc :methods (->> (get-parsed-response)
-                                                     (map #(select-keys % [:namespace :name :snapshotId]))
-                                                     (group-by (juxt :namespace :name))
-                                                     (utils/map-values (partial map :snapshotId))))
-                    ;; FIXME: :error-message is unused
-                    (swap! state assoc :error-message status-text)))}))
+   (fn [{:keys [this]}]
+     (this :-load-validated-method-config))
+   :component-did-update
+   (fn [{:keys [state]}]
+     (let [{:keys [methods loaded-config]} @state]
+       (when (and (not methods) loaded-config)
+         (let [;methodConfiguration (:methodConfiguration loaded-config)
+               ;methodRepoMethod (:methodRepoMethod methodConfiguration)
+               {:keys [methodName methodNamespace]} (get-in loaded-config [:methodConfiguration :methodRepoMethod])]
+           (endpoints/call-ajax-orch
+            {:endpoint (endpoints/list-method-snapshots methodNamespace methodName)
+             :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                        (let [response (get-parsed-response)]
+                          (if success?
+                            (swap! state assoc :methods-response response :methods (->> response
+                                                                                        (group-by (juxt :namespace :name))
```

Comment:
do you still need the group-by now that the endpoint is returning just the relevant namespace/name?

Reply:
removing